detection-rules/impersonation_ms_teams_invite.yml

```diff
@@ -4,29 +4,45 @@ type: "rule"
 severity: "high"
 source: |
   type.inbound
-  and strings.icontains(body.current_thread.text, 'Microsoft Teams')
-  and strings.icontains(body.current_thread.text, 'join the meeting now')
-  and strings.contains(body.current_thread.text, 'Meeting ID:')
-  and strings.contains(body.current_thread.text, 'Passcode:')
-  
+  and (
+    (
+      strings.icontains(body.current_thread.text, 'Microsoft Teams')
+      and strings.icontains(body.current_thread.text, 'join the meeting now')
+      and strings.contains(body.current_thread.text, 'Meeting ID:')
+      and strings.contains(body.current_thread.text, 'Passcode:')
+    )
+    or 3 of (
+      strings.icontains(body.current_thread.text, "internal"),
+      strings.icontains(body.current_thread.text, "teams"),
```

Comment:
the body can contain "internal", "message", and "meeting"? 

should 1 of these be "required"